### PR TITLE
fix: return only public properties for form

### PR DESCRIPTION
**Pull request by jeberhardt**
# Summary | Résumé

There is no issue for this PR.

The `getTemplateByID` request being made for the forms page was returning non-public form config data. This change sanitizes the data by calling `onlyIncludePublicProperties()` on the form, returning only the properties that can safely be sent to the browser when being requested by an unauthorized user (ie: someone filling out the form).

Screen Shot 2022-09-14 at 2 32 28 PM

# Test instructions | Instructions pour tester la modification

- Load any form into the browser, as if you were going to fill it out.
- Observe in the browser Developer Tools / Network tab, that the data being sent from the server does not include non-public properties, such as `submission`, or `internalTitleEN`, among other such properties. 
- It should include default values for `displayAlphaBanner` and `securityAttribute`, even if they are not present in the formConfig saved in the database

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [ ] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [ ] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [ ] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [x] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [ ] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

**Review by craigzour — APPROVED** [review]

**Review by Moro-Code — CHANGES_REQUESTED** [review]
minor comment

File: pages/id/[form]/[[...step]].tsx [hunk 1]
```diff
@@ -73,6 +73,9 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
     if (!formID || Array.isArray(formID)) return redirect(context.locale);
 
     form = await getTemplateByID(formID);
+    if (form) {
+      form = onlyIncludePublicProperties(form);
```
[the remark this answers is not in the document]
Comment (jeberhardt) [reply pages/id/[form]/[[...step]].tsx:77]: If form is `null`, as could be the case after `getTemplateByID`, then `onlyIncludePublicProperties` would throw an error, which I don't believe would be caught anywhere.

[the remark this answers is not in the document]
Comment (Moro-Code) [reply pages/id/[form]/[[...step]].tsx:77]: It wouldn't be null 

```JS
if (
    process.env.APP_ENV !== "test" &&
    (!form || (!form?.formConfig?.publishingStatus && !unpublishedForms))
  ) 
 ```
 we check this on line 83

**Review by craigzour — APPROVED** [review]

File: pages/id/[form]/[[...step]].tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (jeberhardt) [reply pages/id/[form]/[[...step]].tsx]: I have to move this back, as it is breaking all of our **Cypress** tests, since **APP_ENV==="test"**.

**Comment by bryan-robitaille** [comment]
The Cypress tests were failing because `getServerSideProps` was passing an undefined object to `onlyIncludePublicProperties` which caused the server-side code to crash.
I refactored and added in Typescript declarations so it would be easier to see.

**Review by Moro-Code — APPROVED** [review]
Looks good to merge now

**Pull request merged**